### PR TITLE
Add regression tests for Snapshot$data empty-collection merge

### PR DESCRIPTION
**Pull request by Felixmil**
Issue #59 reported that emptying a building-block collection via `remove_*()` would silently re-introduce the original entries on export because the per-section `if (length(...) > 0)` guard in `Snapshot$data` skipped the merge. PR #58 already addressed this when it refactored `Snapshot$data` into a per-section adapter table with a tri-state contract (cache `NULL` replays original, cache `list()` exports empty, cache non-empty rebuilds from items). This PR confirms the fix with regression tests against the three lazy-cache states and notes the cross-reference in NEWS.

## Tests

The new file `tests/testthat/test-snapshot-empty-merge.R` exercises:

- Scenario A (load N, remove all): Compounds, Individuals, ObservedData.
- Scenario B (untouched empty or missing section): Compounds.
- Scenario C (read-only access): Compounds, Individuals.

Compounds and Individuals stand in for the seven sections that share `.default_export_adapter`. ObservedData has its own `.observed_data_export_adapter` and is covered separately for scenario A.

## NEWS

The existing #35 bug-fix bullet is extended to also reference #59, since the same fix closes both issues.

Closes #59

### CODE REVIEW

File: tests/testthat/test-snapshot-empty-merge.R [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# honour user intent across the three lazy-cache states (NULL untouched,
+# `list()` cleared, non-empty hydrated). PR #58 introduced the tri-state
+# adapter contract that fixed the pre-existing merge bug; these tests guard
+# against future regressions.
+
+# Scenario A: load N items, remove every one via the matching mutator. The
+# exported section must be empty (the user's clear intent), not the original
+# items resurrected.
+
+test_that("remove_compound clearing all entries drops the Compounds section on export", {
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply tests/testthat/test-snapshot-empty-merge.R]: Good catch. Renamed to "empties" since the JSON keeps the `Compounds` key with an empty list (unlike `ObservedData`, which the adapter drops by returning NULL). Fixed in 6614cb0.

File: tests/testthat/test-snapshot-empty-merge.R [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  snapshot <- load_snapshot(test_path("data", "test_snapshot.json"))
+  for (name in names(snapshot$compounds)) {
+    snapshot$remove_compound(name)
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply tests/testthat/test-snapshot-empty-merge.R]: Agreed, the loop was redundant. `remove_compound` already accepts a character vector. Replaced with the single vectorized call to mirror the `remove_observed_data` test below. Fixed in 6614cb0.

File: tests/testthat/test-snapshot-empty-merge.R [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  parsed <- jsonlite::fromJSON(
+    out,
+    simplifyVector = FALSE,
+    simplifyDataFrame = FALSE
+  )
+  expect_length(parsed$Compounds, 0)
+})
+
+test_that("remove_individual clearing all entries drops the Individuals section on export", {
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply tests/testthat/test-snapshot-empty-merge.R]: Same fix as for Compounds: renamed to "empties" since `Individuals` stays in the JSON as `[]`. Fixed in 6614cb0.

File: tests/testthat/test-snapshot-empty-merge.R [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  snapshot <- load_snapshot(test_path("data", "test_snapshot.json"))
+  for (name in names(snapshot$individuals)) {
+    snapshot$remove_individual(name)
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply tests/testthat/test-snapshot-empty-merge.R]: Done, vectorized to match `remove_observed_data` on line 49. Fixed in 6614cb0.
